### PR TITLE
carbon-factory v5 upgrade

### PR DESCRIPTION
**Pull request by andrewjtait**

### CODE REVIEW

File: src/components/carousel/carousel.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ class Carousel extends React.Component {
 
     if (newIndex === currentIndex) {
       return;
-    } else if (newIndex > currentIndex) {
+    } if (newIndex > currentIndex) {
```
Comment (andrewjtait) [inline src/components/carousel/carousel.js]: err - not sure about this?

Comment (SeanArmstrong) [reply src/components/carousel/carousel.js]: huh

Comment (SeanArmstrong) [reply src/components/carousel/carousel.js]: Maybe if it was
```
if (newIndex === currentIndex) { return; }
if (newIndex > currentIndex) {
  this.transitionDirection = NEXT;
} else {
  this.transitionDirection = PREVIOUS;
```

Comment (andrewjtait) [reply src/components/carousel/carousel.js]: yeah maybe it is ok - it highlights that the else is unecessary

File: src/components/dropdown-filter-ajax/dropdown-filter-ajax.js [hunk 2]
```diff
@@ -158,9 +158,11 @@ class DropdownFilterAjax extends DropdownFilter {
    */
   handleBlur = () => {
     if (!this.blockBlur) {
-      const filter = this.props.create ? this.state.filter : null;
       // close list and reset filter
-      this.setState({ open: false, filter });
+      this.setState(prevState => ({
+        open: false,
+        filter: this.props.create ? prevState.filter : null
```
Comment (SeanArmstrong) [inline src/components/dropdown-filter-ajax/dropdown-filter-ajax.js]: Is this just because it was a ternary operator?

File: src/components/grouped-character/grouped-character.js [hunk 3]
```diff
@@ -139,12 +139,12 @@ const GroupedCharacter = Input(InputLabel(InputValidation(
 
   isValidKeypress = (ev) => {
     return (
-      !Events.isNumberKey(ev) &&
-      !Events.isAlphabetKey(ev) &&
-      !Events.isTabKey(ev) &&
-      !Events.isDeleteKey(ev) &&
-      !Events.isBackspaceKey(ev) &&
-      !Events.isNavigationKey(ev)
+      !Events.isNumberKey(ev)
```
Comment (SeanArmstrong) [inline src/components/grouped-character/grouped-character.js]: I prefer the previous

File: src/components/dropdown-filter-ajax/dropdown-filter-ajax.js (hunk 2 above)
Comment (andrewjtait) [reply src/components/dropdown-filter-ajax/dropdown-filter-ajax.js]: It is bad to `setState` with existing state as it is not always a synchronous event, so we needed to use the function version.

File: src/components/grouped-character/grouped-character.js (hunk 3 above)
Comment (andrewjtait) [reply src/components/grouped-character/grouped-character.js]: I'm indifferent - will go with whatever people prefer. See https://eslint.org/docs/rules/operator-linebreak

Comment (andrewjtait) [reply src/components/grouped-character/grouped-character.js]: One advantage to having them down the left means you can look straight down the left edge to see the operators.

**Comment by SeanArmstrong** [comment]
Change is scary but 👍

File: src/components/grouped-character/grouped-character.js (hunk 3 above)
Comment (andrewjtait) [reply src/components/grouped-character/grouped-character.js]: I tried switching it and comparing the diff, it is actually easier to understand whats happening with the operator on the left - otherwise it gets lost at the end of the line.

**Comment by andrewjtait** [comment]
Tested components with significant changes within the demo site ✅
